Library/OcCryptoLib/Sha2.c

```diff
@@ -324,11 +324,19 @@ Sha256 (
 //
 // Sha 512 functions
 //
+VOID
+EFIAPI
+Sha512TransformAvx (
+  IN OUT UINT64      *State,
+  IN     CONST UINT8 *Data,
+  IN     UINTN       BlockNb
+  );
+
 VOID
 Sha512Transform (
-  SHA512_CONTEXT  *Context,
-  CONST UINT8     *Data,
-  UINTN           BlockNb
+  UINT64      *State,
+  CONST UINT8 *Data,
+  UINTN       BlockNb
```

Comment:
Would still suggest updating the original `Sha512Transform` signature completely, to match the now more carefully constructed signature of your new method, including use of `IN`, `OUT`, here and in the method definition.